aptos-move/framework/aptos-framework/sources/coin.move

```diff
@@ -97,6 +113,51 @@ module aptos_framework::coin {
     /// Capability required to burn coins.
     struct BurnCapability<phantom CoinType> has copy, store { }
 
+    //
+    // Supply functions
+    //
+
+    /// Creates new supply which can be an aggregator (parallelizable) or a standard
+    /// integer.
+    fun initialize_supply(parallelizable: bool): Supply {
+        Supply {
+            inner: optional_aggregator::new(MAX_U128, parallelizable),
+        }
+    }
+
+    /// Adds `amount` to total supply of `CoinType`. Called when minting coins.
+    fun add(supply: &mut Supply, amount: u128) {
```

Comment:
Do we need all of these wrapper functions? They're only called once. May be only read is worth keeping. Should rename to get_total_supply or something

Reply:
Good point, I was also thinking about that:)